src/client/Microsoft.Identity.Client/TelemetryCore/Internal/HttpTelemeteryContent.cs

```diff
@@ -0,0 +1,131 @@
+
+            CorrelationId.Add(mostRecentStoppedApiEvent[MsalTelemetryBlobEventNames.MsalCorrelationIdConstStrKey]);
+            ApiId.Add(mostRecentStoppedApiEvent[MsalTelemetryBlobEventNames.ApiIdConstStrKey]);
+
+            if (mostRecentStoppedApiEvent.ContainsKey(MsalTelemetryBlobEventNames.ApiErrorCodeConstStrKey))
+            {
+                ErrorCode.Add(mostRecentStoppedApiEvent[MsalTelemetryBlobEventNames.ApiErrorCodeConstStrKey]);
+            }
+
+            string apiIdCorIdData = CreateApiIdAndCorrelationIdContent();
+            // csv expected format:
+            // 2|silent_successful_count|failed_requests|errors|platform_fields
+            // failed_request can be further expanded to include:
+            // api_id_1,correlation_id_1,api_id_2,correlation_id_2|error_1,error_2
+            string data = string.Format(CultureInfo.InvariantCulture,
+                $"{TelemetryConstants.HttpTelemetrySchemaVersion2}{TelemetryConstants.HttpTelemetryPipe}{successfulSilentCallCount}{TelemetryConstants.HttpTelemetryPipe}" +
+                $"{CreateFailedRequestsContent(apiIdCorIdData)}" +
+                $"{TelemetryConstants.HttpTelemetryPipe}");
+
+            if (data.Length > 3800)
+            {
+                ClearData();
+                return string.Empty;
+            }
+
+            return data;
+        }
+
+        public string GetCsvAsCurrent(ConcurrentDictionary<EventKey, EventBase> eventsInProgress)
+        {
+            if (eventsInProgress == null)
+            {
+                return string.Empty;
+            }
+
+            IEnumerable<KeyValuePair<EventKey, EventBase>> apiEvent = eventsInProgress.Where(e => e.Key.EventName == "msal.api_event");
+
+            foreach (KeyValuePair<EventKey, EventBase> events in apiEvent)
+            {
+                events.Value.TryGetValue(MsalTelemetryBlobEventNames.ApiIdConstStrKey, out string apiId);
+                ApiId.Add(apiId);
+                events.Value.TryGetValue(MsalTelemetryBlobEventNames.ForceRefreshId, out string forceRefresh);
+                _forceRefresh = forceRefresh;
+            }
+
+            string apiEvents = string.Join(",", ApiId);
+
+            // csv expected format:
+            // 2|api_id,force_refresh|platform_config
+            string[] myValues = new string[] {
+                apiEvents,
+                ConvertFromStringToBitwise(_forceRefresh)};
+
+            string csvString = string.Join(",", myValues);
+            return $"{TelemetryConstants.HttpTelemetrySchemaVersion2}{TelemetryConstants.HttpTelemetryPipe}{csvString}{TelemetryConstants.HttpTelemetryPipe}";
+        }
+
+        private string CreateApiIdAndCorrelationIdContent()
+        {
+            List<string> combinedApiIdCorrId = new List<string>();
+            List<string> apiId = new List<string>();
+            apiId = ApiId;
+
+            foreach (var apiIds in ApiId)
+            {
+                combinedApiIdCorrId.Add(apiIds);
```

Comment:
Why do we need to combine these 2 unrelated IDs?

Reply:
the schema format for this section is: |api_id1, corr_id1, api_id2, corr_id2|

---
In reply to: [389792841](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1693#discussion_r389792841) [](ancestors = 389792841)